testing/sdk_security.py

```diff
@@ -129,17 +129,19 @@ def get_permissions(service_account_name: str, role: str, linux_user: str) -> Li
         }]
 
 
-def grant_permissions(linux_user: str, role_name: str, service_account_name: str) -> None:
+def grant_permissions(linux_user: str, role_name: str, service_account_name: str, additional_permissions: List[dict]) -> None:
     log.info("Granting permissions to {account}".format(account=service_account_name))
-    permissions = get_permissions(service_account_name, role_name, linux_user)
+    default_permissions = get_permissions(service_account_name, role_name, linux_user)
+    permissions = default_permissions + additional_permissions
     for permission in permissions:
         _grant(**permission)
     log.info("Permission setup completed for {account}".format(account=service_account_name))
 
 
-def revoke_permissions(linux_user: str, role_name: str, service_account_name: str) -> None:
+def revoke_permissions(linux_user: str, role_name: str, service_account_name: str, additional_permissions: List[dict]) -> None:
```

Comment:
Should we not keep track of the permissions that were granted and instead pass them in to `revoke_permissions` instead of constructing the list of permissions again?

Reply:
I agree, let's pass them as a parameter